system/jlib/jsocket.cpp

```diff
@@ -893,6 +894,8 @@ int CSocket::post_connect ()
         nagling = true;
         set_nagle(false);
         state = ss_open;
+        char lname[256];
+        localPort = name(lname, sizeof(lname), false);
```

Comment:
a slight waste getting lname when not needed.
CSocket::name() should fill in a StringBuffer really too, which it does internally, only to copy to the fixed sized passed in buffer.
You could change it to 2 methods:
```
void CSocket::getEndpoint(SocketEndpoint &ep); 
unsigned CSocket::getName(StringBuffer &name);
```

and use getEndpoint() here.


Same is true of CSocket::peer_name()